src/stores/auth.js

```diff
@@ -117,10 +116,12 @@ export const useAuthStore = defineStore('auth', {
           const { gameId } = route.params;
           gameStore.requestSpectate(Number(gameId));
         }
-
         if (gameId && (isGame || isLobby)) {
           await this.requestReauthenticate({ username }).then(({ game }) => {
             gameStore.updateGame(game);
+            if (+router.currentRoute.value.params.gameId !== game.id) {
```

Comment:
Let's use Number() to cast for consistency as that's how we're doing it elsewhere